iree/compiler/Conversion/LinalgToLLVM/ConvImg2ColMatmulConversion.cpp

```diff
@@ -0,0 +1,180 @@
+    indexingMaps.emplace_back(AffineMap::getMultiDimIdentityMap(
+        ColBufferMemrefType.getRank(), rewriter.getContext()));
+
+    rewriter.create<linalg::GenericOp>(
+        loc, ArrayRef<Type>{}, linalgOpArgs,
+        1,  // args_in
+        1,  // args_out
+        indexingMaps, loopAttributeTypes,
+        [&](OpBuilder &nestedBuilder, Location nestedLoc, ValueRange args) {
+          nestedBuilder.create<linalg::YieldOp>(nestedLoc, args[0]);
+        });
+
+    auto getIndicesVector = [](int start, int end) {
+      SmallVector<int64_t, 2> indices;
+      for (int i = start; i <= end; ++i) {
+        indices.push_back(i);
+      }
+      return indices;
+    };
+
+    // n, d1, d2,....dn
+    auto batchAndSpatialIndices =
+        getIndicesVector(0, op.getNumSpatialDimensions());
+    auto featureIndices = getIndicesVector(op.getNumSpatialDimensions() + 1,
+                                           op.getNumSpatialDimensions() + 3);
+
+    SmallVector<linalg::ReassociationIndices, 4> lhsCollapsedDimsList = {
+        batchAndSpatialIndices, featureIndices};
+    SmallVector<linalg::ReassociationIndices, 4> rhsCollapsedDimsList = {
+        batchAndSpatialIndices, {op.getNumSpatialDimensions() + 1}};
+
+    SmallVector<linalg::ReassociationIndices, 4> resultCollapsedDimsList = {
+        batchAndSpatialIndices, {op.getNumSpatialDimensions() + 1}};
+
+    auto reshapedColBufferType =
+        MemRefType::get({spatialSize, filterSpatialSize * inputFeatures},
+                        filterShapeType.getElementType());
+
+    auto reshapedfilterType =
+        MemRefType::get({filterSpatialSize * inputFeatures, outputFeatures},
+                        filterShapeType.getElementType());
+
+    auto reshapedOutputType = MemRefType::get({spatialSize, outputFeatures},
+                                              filterShapeType.getElementType());
+
+    Value reshapedLhs = rewriter.create<linalg::ReshapeOp>(
+        loc, reshapedColBufferType, result, lhsCollapsedDimsList);
+
+    Value reshapedRhs = rewriter.create<linalg::ReshapeOp>(
+        loc, reshapedfilterType, op.filter(), rhsCollapsedDimsList);
+
+    Value reshapedResult = rewriter.create<linalg::ReshapeOp>(
+        loc, reshapedOutputType, op.output(), resultCollapsedDimsList);
+
+    rewriter.create<linalg::MatmulOp>(
+        loc, ArrayRef<Value>{reshapedLhs, reshapedRhs}, reshapedResult);
+
+    rewriter.eraseOp(op);
+    return success();
+  }
+};
+
+struct ConvImg2ColMatmulConversionPass
+    : PassWrapper<ConvImg2ColMatmulConversionPass, FunctionPass> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<linalg::LinalgDialect>();
+  }
+  void runOnFunction() override;
+};
+}  // namespace
+
+void ConvImg2ColMatmulConversionPass::runOnFunction() {
+  auto funcOp = getOperation();
+  auto context = funcOp.getContext();
+  OwningRewritePatternList patterns;
+
+  patterns.insert<ConvImg2ColMatmulConversion>(context);
```

Comment:
Could you move this into a function, say `populateIm2ColMatmulConversionPattern` and expose it as a way to import this pattern.